python_tests/test_cli.py

```diff
@@ -0,0 +1,120 @@
+import tempfile
+
+from unittest.mock import patch, MagicMock
+
+from optuna_dashboard._cli import main
+
+
+def test_cli_args_override_config_file() -> None:
+    """Test that CLI arguments override config file values."""
+    config_content = """
+[optuna_dashboard]
+storage = "sqlite:///config.db"
+port = 9000
+host = "0.0.0.0"
+server = "gunicorn"
+"""
+    with tempfile.NamedTemporaryFile(mode="w", suffix=".toml") as f:
+        f.write(config_content)
+        f.flush()
+
+        with patch("optuna_dashboard._cli.get_storage") as mock_get_storage, patch(
+            "optuna_dashboard._cli.create_app"
+        ) as mock_create_app, patch(
+            "optuna_dashboard._cli.run_debug_server"
+        ) as mock_run_debug_server, patch("optuna_dashboard._cli.DEBUG", True):
+            # Setup mocks to prevent actual execution.
+            mock_storage = MagicMock()
+            mock_get_storage.return_value = mock_storage
+            mock_app = MagicMock()
+            mock_create_app.return_value = mock_app
+            mock_run_debug_server.return_value = None  # Prevent server startup.
```

Comment:
Executing `main()` in `cli.py` involves patching many components, making the test code complicated. From a maintainability perspective, this could potentially increase future engineering costs.

How about adding a helper function or data class that determines values according to the priority of CLI options and configuration files?